crates/orchestrator/src/generators/identity.rs

```diff
@@ -0,0 +1,39 @@
+use hex::FromHex;
+use libp2p::identity::{ed25519, Keypair};
+use sha2::digest::Digest;
+
+use super::errors::GeneratorError;
+
+pub fn generate_for_node(name: &str) -> Result<(String, String), GeneratorError> {
+    let key = hex::encode(sha2::Sha256::digest(name));
+
+    let bytes = <[u8; 32]>::from_hex(key.clone()).map_err(|_| {
+        GeneratorError::IdentityGeneration("can not transform hex to [u8;32]".into())
+    })?;
+    let sk = ed25519::SecretKey::try_from_bytes(bytes)
+        .map_err(|_| GeneratorError::IdentityGeneration("can not create sk from bytes".into()))?;
+    let local_identity: Keypair = ed25519::Keypair::from(sk).into();
+    let local_public = local_identity.public();
+    let local_peer_id = local_public.to_peer_id();
+
+    Ok((key, local_peer_id.to_base58()))
+}
```

Comment:
From what I understand, we do node_name -> sha256 -> hex -> key ? It's been far since I didn't manipulate substrate